cuda/base/types.hpp

```diff
@@ -124,6 +121,17 @@ struct culibs_type_impl<std::complex<double>> {
     using type = cuDoubleComplex;
 };
 
+
+template <>
+struct culibs_type_impl<half> {
+    using type = __half;
+};
+
+template <>
+struct culibs_type_impl<std::complex<half>> {
+    using type = __half2;
```

Comment:
Just for my understanding, `__half2` is a vector type derived from `__half`?

Reply:
you are right. only one function `__hcmadd` does complex operation, but the other are all vector type operation.

Reply:
I have check cusparse they use __half2 to hold complex<__half>, too. https://docs.nvidia.com/cuda/cusparse/#cudadatatype-t